test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/MemberExpressionCacheKeyComparerTest.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq.Expressions;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures
+{
+    public class MemberExpressionCacheKeyComparerTest
```

Comment:
Add tests covering differences later in a chained expression e.g. `GetKey(m => m.TestModel2.Name) != GetKey(m => m.TestModel2.Id)` and `GetKey(m => m.DateTime.Day) != GetKey(m => m.DateTime.Month)`

Reply:
thx